File: El Mamoun/Lv 1/Problem 11.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
 Write a program to ask the user to enter: • Mark1, Mark2, Mark3
 Then Print the Average of entered Marks, and print “PASS” if average>=50, otherwise print “FAIL”
*/

namespace Problem_11
{
    internal class Program
    {
        struct stMatks
        {
            public int Mark1;
            public int Mark2;
            public int Mark3;
        }
        static stMatks ReadNumbers()
        {
            stMatks Marks;

            Console.Write("Please Enter Mark1?: ");
            Marks.Mark1 = Convert.ToInt32(Console.ReadLine());

            Console.Write("Please Enter Mark2?: ");
            Marks.Mark2 = Convert.ToInt32(Console.ReadLine());

            Console.Write("Please Enter Mark3?: ");
            Marks.Mark3 = Convert.ToInt32(Console.ReadLine());

            return Marks;
        }
        static int SumOf3Numbers(stMatks Marks)
        {
            return Marks.Mark1 + Marks.Mark2 + Marks.Mark3;
        }
        static float Average3Numbers(stMatks Marks)
        {
            return (float)(SumOf3Numbers(Marks) / 3);
        }
        static bool CheckPassOrFail(stMatks Marks)
        {
            return (Average3Numbers(Marks) >= 50);
        }
        static void PrintResult(stMatks Marks)
        {
            Console.WriteLine(Average3Numbers(Marks));

            if (CheckPassOrFail(Marks))
                Console.Write("Pass");
            else
                Console.Write("Fail");
        }
     
        static void Main(string[] args)
        {

            PrintResult(ReadNumbers());
            Console.ReadKey();
        }
    }
}

File: El Mamoun/Lv 1/Problem 11.cpp
#include<iostream>
using namespace std;

/*
 Write a program to ask the user to enter: • Mark1, Mark2, Mark3
 Then Print the Average of entered Marks, and print “PASS” if average>=50, otherwise print “FAIL”
*/

enum enPassOrFail { Pass = 1, Fail=2 };

void ReadNumbers(int& Num1, int& Num2, int& Num3)
{

	cout << "\nPlease Enter Your First Number? ";
	cin >> Num1;

	cout << "\nPlease Enter Your Second Number? ";
	cin >> Num2;

	cout << "\nPlease Enter Your Third Number? ";
	cin >> Num3;
}

int SumOf3Numbers(int Num1, int Num2, int Num3)
{
	return Num1 + Num2 + Num3;
}

float Average3Number(int Num1, int Num2, int Num3)
{
	return float(SumOf3Numbers(Num1, Num2, Num3) / 3);
}

enPassOrFail CheckAverage(float Average)
{

	if (Average >= 50)
		return enPassOrFail::Pass;

	else
		return enPassOrFail::Fail;
}

void PrintResults(float Average)
{

	cout << "\nThe Average = " << Average << endl;

	if (CheckAverage(Average) == enPassOrFail::Pass)
		cout << "Pass";
	else
		cout << "Fail";
}


int main()
{

	int Num1, Num2, Num3;
	
	ReadNumbers(Num1, Num2, Num3);
	PrintResults(Average3Number(Num1, Num2, Num3));

	return 0;
}